example/lib/test_scenarios/sensor_tag_test_scenario.dart

```diff
@@ -179,4 +210,28 @@ class SensorTagTestScenario {
     await peripheral.disconnectOrCancelConnection();
     log("Disconnected!");
   }
+
+  void _startMonitoringTemperature(
+      Stream<Uint8List> characteristicUpdates, Function log) {
+    log("START TEMPERATURE MONITORING");
+    monitoringStreamSubscription?.cancel();
+    monitoringStreamSubscription = characteristicUpdates
+        .map(_convertToTemperature)
+        .where((temperature) => temperature != 0)
+        .take(1)
```

Comment:
Why take(1)? This method should return stream subscription which should be disposed at the end of the test method

Reply:
It's just for the sake of logs clarity. The update interval is really fast and it shows duplicated values in a really short period.Even if you turn the update off, the temperature characteristic will be notified about update with temperature value set to 0 (it will be done multiple times in a fast way). Take(1) is to prevent from spamming in our logs